libs/brain/switch/src/lib/brn-switch.component.ts

```diff
@@ -141,33 +175,23 @@ export class BrnSwitchComponent implements AfterContentInit, OnDestroy {
 
 	constructor() {
 		effect(() => {
-			/** search for the label and set the disabled state */
-			let parent = this._renderer.parentNode(this._elementRef.nativeElement);
-			if (!parent) return;
-			// if parent is a HLM-SWITCH, then we need to go up one more level to get the label
-			if (parent?.tagName === 'HLM-SWITCH') {
-				parent = this._renderer.parentNode(parent);
-			}
-			if (!parent) return;
-			// check if parent is a label and assume it is for this checkbox
-			if (parent?.tagName === 'LABEL') {
-				this._renderer.setAttribute(parent, 'data-disabled', this.state().disabled() ? 'true' : 'false');
-				return;
-			}
-			if (!this._isBrowser) return;
-
-			const label = parent?.querySelector(`label[for="${this.forChild(this.state().id)}"]`);
-			if (!label) return;
-			this._renderer.setAttribute(label, 'data-disabled', this.state().disabled() ? 'true' : 'false');
+			const isDisabled = this.state().disabled();
+			const state = this.state();
+			if (!this._elementRef.nativeElement || !this._isBrowser) return;
+
+			const switchElement = this._elementRef.nativeElement;
+
+			const labelElement = this._findLabelForSwitch(switchElement);
```

Comment:
Rather than needing this `_findLabelForSwitch` method, could we just use `this._elementRef.nativeElement.closest('label')` to search the parents for a label?